### PR TITLE
[Fix] compressed-tensors block FP8: requantize weight scales to UE8M0 for DeepGEMM on Blackwell

### PR DESCRIPTION
**Pull request by Jiminator**
## Motivation

Block-wise FP8 checkpoints loaded through the compressed-tensors path (`CompressedTensorsW8A8Fp8`, `BLOCK` strategy) produce NaN / empty output on Blackwell (SM100), making such models unusable on Blackwell with the default GEMM backend (e.g. `poolside/Laguna-M.1-FP8`: GSM8K 0.00%, all generations empty).

On Blackwell the block-FP8 linear dispatches to DeepGEMM, which quantizes activations to UE8M0 scales and expects the weight scales to be UE8M0-packed as well. `Fp8LinearMethod` already requantizes the weight scales to UE8M0 (via `requant_weight_ue8m0_inplace`) when DeepGEMM is the active runner, but the `BLOCK` branch of `CompressedTensorsW8A8Fp8.process_weights_after_loading` never did — it kept the raw float32 block scales. DeepGEMM then combined UE8M0 activation scales with float32 weight scales and emitted NaN logits.

(NVFP4 is unaffected — CUTLASS path; the compressed-tensors FP8 MoE experts are unaffected — they do not dispatch DeepGEMM.)

## Modifications

- **compressed-tensors block FP8 fix.** In the `BLOCK` branch of `CompressedTensorsW8A8Fp8.process_weights_after_loading`, requantize the weight scales to UE8M0 when DeepGEMM is the active runner, so DeepGEMM no longer pairs UE8M0 activation scales with float32 weight scales. Non-Blackwell hardware and the `triton` / `cutlass` backends keep float32 scales unchanged.
- **Shared requant helper.** Extract the UE8M0 weight-scale requant decision into a single helper `maybe_requant_block_scale_ue8m0()` in `fp8_utils.py`, and route all three call sites through it — `Fp8LinearMethod`, `Fp8MoEMethod`, and the compressed-tensors scheme — so the gate (DeepGEMM runner selected + 128×128 block + bf16/aligned shape + not-already-UE8M0) lives in one place instead of being duplicated. This is a behavior-preserving refactor of the existing `Fp8LinearMethod` / `Fp8MoEMethod` paths.
- **Tests.** Add a registered CPU unit test `test/registered/unit/layers/quantization/test_deepgemm_ue8m0_requant.py` (mockable; checks the requant decision fires once for a supported DeepGEMM / bf16 / 128×128 layer and skips for non-bf16 output, DeepGEMM-unsupported shape, a non-DeepGEMM runner, and non-128×128 block sizes), plus a Blackwell-only end-to-end regression test `test/manual/quant/test_compressed_tensors_fp8_block_ue8m0.py` (skips unless SM100 + DeepGEMM).

Gated by `should_deepgemm_weight_requant_ue8m0()` (requires SM100 + `DEEPGEMM_SCALE_UE8M0`), so it is a no-op on non-Blackwell hardware and non-DeepGEMM backends.

## Accuracy Tests

`poolside/Laguna-M.1-FP8`, 8×B200, `--tp 8`, default GEMM backend, GSM8K (1319, greedy):

| | before | after |
|---|---|---|
| GSM8K | 0.00% (empty output) | 92.80% |

92.80% matches the same checkpoint on H200 (FP8, 93.25%).

The added unit tests assert the weight scales are requantized to UE8M0 and that the layer output is finite and within tolerance of a bf16 dequant reference (pre-fix: NaN). Passes on B200.

## Speed Tests and Profiling

No new kernels — this only fixes weight-scale preparation at load time, so the steady-state GEMM is unchanged. The fix restores the fast default (DeepGEMM) path on Blackwell; the pre-existing `--fp8-gemm-backend triton` fallback also produces correct output but is slower, so before this fix the only way to run the model produced either NaN (default) or degraded throughput (triton).

`poolside/Laguna-M.1-FP8`, 8×B200, `--tp 8`, `python -m sglang.bench_serving` (random, 1024 in / 512 out), same workload per backend:

| backend | output tok/s | total tok/s | median TTFT | median ITL | single-stream decode tok/s | single-stream TPOT |
|---|---|---|---|---|---|---|
| **default (DeepGEMM)** | **1363.6** | **4088.1** | 790 ms | 20.4 ms | **103.6** | **9.59 ms** |
| `triton` | 1084.2 | 3250.4 | 826 ms | 25.7 ms | 67.0 | 14.87 ms |

The default DeepGEMM path is **+25.8%** output throughput at concurrency 32 and **+54.8%** decode throughput single-stream versus the `triton` fallback. No profiling delta is expected since no kernels change; the win is purely from using DeepGEMM instead of the triton GEMM fallback.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations). (n/a — no user-facing API or behavior change beyond fixing the NaN)
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #28204157956](https://github.com/sgl-project/sglang/actions/runs/28204157956)
Latest PR Test (Extra): :x: [Run #28204157853](https://github.com/sgl-project/sglang/actions/runs/28204157853)

### CODE REVIEW

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 1]
```diff
@@ -1277,6 +1277,43 @@ def requant_weight_ue8m0_inplace(weight, weight_scale_inv, weight_block_size):
     weight_scale_inv.data = new_weight_scale_inv
 
 
+def maybe_requant_block_scale_ue8m0(
+    weight: torch.nn.Parameter,
+    weight_scale: torch.nn.Parameter,
+    weight_block_size: Optional[List[int]],
+    *,
```
Comment (b8zhong) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: We shouldn't use * if possible

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 2]
```diff
@@ -1277,6 +1277,43 @@ def requant_weight_ue8m0_inplace(weight, weight_scale_inv, weight_block_size):
     weight_scale_inv.data = new_weight_scale_inv
 
 
+def maybe_requant_block_scale_ue8m0(
```
Comment (b8zhong) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: Nit: can you rename this function to be a bit more descriptive, (since maybe is hard to trace the reason)

**Review by b8zhong — APPROVED** [review]
